go/private/rules/test.bzl

```diff
@@ -120,6 +120,7 @@ def _go_test_impl(ctx):
         executable = go.toolchain._builder,
         arguments = [arguments],
         toolchain = GO_TOOLCHAIN_LABEL,
+        env = go.env,
```

Comment:
This looks good, but I would like to understand better why it fixes the issue. Which variable do we end up passing in here that matters?

Reply:
The issue I faced was `CGO_ENABLED`. `build.Context` does not properly filter out cgo files when generating testmain when cgo is disabled.